atomic_reactor/util.py

```diff
@@ -1454,30 +1454,21 @@ def get_log_files(self, osbs, build_id):
 
         # Collect logs from server
         try:
-            logs = osbs.get_build_logs(build_id)
+            logs = osbs.get_build_logs(pipeline_run_name)
         except OsbsException as ex:
             self.log.error("unable to get build logs: %s", ex)
             return output
-        except TypeError:
-            # Older osbs-client has no get_orchestrator_build_logs
-            self.log.error("OSBS client does not support get_orchestrator_build_logs")
-            return output
 
-        platform_logs = {}
-        for entry in logs:
-            platform = entry.platform
-            if platform not in platform_logs:
-                filename = 'orchestrator' if platform is None else platform
-                platform_logs[platform] = NamedTemporaryFile(prefix="%s-%s" %
-                                                             (build_id, filename),
-                                                             suffix=".log", mode='r+b')
-            platform_logs[platform].write((entry.line + '\n').encode('utf-8'))
-
-        for platform, logfile in platform_logs.items():
-            logfile.flush()
-            filename = 'orchestrator' if platform is None else platform
-            metadata = self.get_log_metadata(logfile.name, "%s.log" % filename)
-            output.append(Output(file=logfile, metadata=metadata))
+        # OSBS2 TBD logs for specific arches for binary builds
+        filename = 'orchestrator'
+        logfile = NamedTemporaryFile(prefix=f"{pipeline_run_name}-{filename}",
+                                     suffix=".log", mode='r+b')
+        for containers in logs.values():
+            for log_message in containers.values():
+                logfile.write((log_message + '\n').encode('utf-8'))
```

Comment:
This depends on dict iteration order (which *is* deterministic in python >= 3.7), and also on the `get_build_logs()` method inserting taskruns and containers in the right order.

That's probably fine, but it might be good to mention that in a comment (or to change the return value of `get_build_logs()` to something more appropriate, but that would be a bigger change).

Reply:
I don't mind doing the change, I leave this for more discussion from other team members.

Reply:
you should get_info for pipeline run and read         task_runs = pipeline_run['status']['taskRuns']

and use that order

Reply:
I don't think that would change anything here, as it should be already in that order https://github.com/containerbuildsystem/osbs-client/blob/master/osbs/tekton.py#L631

Reply:
it is looped in order, but then it puts it in dict, and entries from dict aren't read in the same order

Reply:
what do you mean that entries from dict aren't read in the same order?
It is first looped in order and put in a dict in that order, and from python 3.6+ it is guaranteed that it will be read in that order or am I missing something?

Reply:
ah ok so the order is guaranteed by insertion order

Reply:
@chmeliik so as agreed that no changes will be done to get_build_logs() itself, I added a comment here.